### PR TITLE
gcc: statically link against gmp/mpfr/mpc/isl

### PR DESCRIPTION
**Pull request by sxprophet**
Let `gcc` link statically against `gmp/mpfr/mpc/isl` so that we can rebuild `gmp`, etc, even with `gcc`.
Do you have better disposition with the dependencies? For instance, could we just `brew` them into a temp directory and build statically?

### CODE REVIEW

File: Library/Formula/gcc.rb [hunk 1]
```diff
@@ -38,13 +38,25 @@ def osmajor
...
+    sha1 "#{Formula["mpfr"].stable.checksum}"
+  end
+  resource "libmpc" do
+    url "#{Formula["libmpc"].stable.url}"
+    sha1 "#{Formula["libmpc"].stable.checksum}"
+  end
+  resource "isl" do
+    url "#{Formula["isl"].stable.url}"
+    sha1 "#{Formula["isl"].stable.checksum}"
+  end
```
Comment (jacknagel) [inline Library/Formula/gcc.rb]: Expand the URLs and checksums rather than reach into formula objects from the DSL.

**Comment by jacknagel** [comment]
I don't see this as being super important. Why can't you build this stuff with clang?

**Comment by mikemcquaid** [comment]
@jacknagel Although I don't particularly see the use in this case I'd love to see us do this for e.g. `ghc` to remove the `gmp` dependency from everything built with it.

**Comment by jacknagel** [comment]
Passing on this.

**Pull request closed without merging**